db/migrate/20210630000200_create_profile_declarations.rb

```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+class CreateProfileDeclarations < ActiveRecord::Migration[6.1]
+  def change
+    create_table :profile_declarations, id: :uuid do |t|
+      t.references :participant_declaration, null: false, index: true
+      t.references :lead_provider, null: false, index: true
+      t.references :declarable, polymorphic: true, index: true
```

Comment:
No likey because we can't have database foreign key constraints

https://stackoverflow.com/questions/922184/why-can-you-not-have-a-foreign-key-in-a-polymorphic-association/922341#922341

I wasn't aware that the use of `delegated_type` would result in this :disappointed:

Reply:
Yes. It seems to just put some formalisation around referencing polymorphic connections.